sdk/maps/azure-maps-elevation/src/main/java/com/azure/maps/elevation/ElevationClientBuilder.java

```diff
@@ -0,0 +1,386 @@
+     *
+     * @param retryPolicy the retryPolicy value.
+     * @return the ElevationClientBuilder.
+     */
+    public ElevationClientBuilder retryPolicy(RetryPolicy retryPolicy) {
+        this.retryPolicy = Objects.requireNonNull(retryPolicy, "'retryPolicy' cannot be null.");
+        return this;
+    }
+
+    /**
+     * Sets The client options such as application ID and custom headers to set on a request.
+     *
+     * @param clientOptions the clientOptions value.
+     * @return the ElevationClientBuilder.
+     */
+    @Override
+    public ElevationClientBuilder clientOptions(ClientOptions clientOptions) {
+        this.clientOptions = Objects.requireNonNull(clientOptions, "'clientOptions' cannot be null.");
+        return this;
+    }
+
+    /**
+     * Adds a custom Http pipeline policy.
+     *
+     * @param customPolicy The custom Http pipeline policy to add.
+     * @return the ElevationClientBuilder.
+     */
+    @Override
+    public ElevationClientBuilder addPolicy(HttpPipelinePolicy customPolicy) {
+        pipelinePolicies.add(Objects.requireNonNull(customPolicy, "'customPolicy' cannot be null."));
+        return this;
+    }
+
+    /**
+     * Sets the {@link TokenCredential} used to authenticate HTTP requests.
+     *
+     * @param tokenCredential {@link TokenCredential} used to authenticate HTTP requests.
+     * @return The updated {@link ElevationClientBuilder} object.
+     * @throws NullPointerException If {@code tokenCredential} is null.
+     */
+    @Override
+    public ElevationClientBuilder credential(TokenCredential tokenCredential) {
+        this.tokenCredential = Objects.requireNonNull(tokenCredential, "'tokenCredential' cannot be null.");
+        return this;
+    }
+
+    /**
+     * Sets the {@link AzureKeyCredential} used to authenticate HTTP requests.
+     *
+     * @param keyCredential The {@link AzureKeyCredential} used to authenticate HTTP requests.
+     * @return The updated {@link ElevationClientBuilder} object.
+     * @throws NullPointerException If {@code keyCredential} is null.
+     */
+    @Override
+    public ElevationClientBuilder credential(AzureKeyCredential keyCredential)  {
+        this.keyCredential = Objects.requireNonNull(keyCredential, "'keyCredential' cannot be null.");
+        return this;
+    }
+
+    /**
+     * Sets retry options
+     * @param retryOptions the retry options for the client
+     * @return a reference to this {@code ElevationClientBuilder}
+     */
+    @Override
+    public ElevationClientBuilder retryOptions(RetryOptions retryOptions) {
+        this.retryOptions = retryOptions;
+        return this;
+    }
+
+    /**
+     * Builds an instance of ElevationClientImpl with the provided parameters.
+     *
+     * @return an instance of ElevationClientImpl.
+     */
+    private ElevationClientImpl buildInnerClient() {
+        if (endpoint == null) {
+            this.endpoint = "https://atlas.microsoft.com";
+        }
```

Comment:
Why does `endpoint` have a default value? Also, building shouldn't mutate the builder but instead use local variables.

Reply:
That is the default endpoint for Azure Maps